content/en/llm_observability/setup/sdk/nodejs.md

```diff
@@ -0,0 +1,713 @@
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function sanitizeInput () {
+  ... // user application logic
+  return
+}
+sanitizeInput = llmobs.wrap('task', sanitizeInput)
+{{< /code-block >}}
+
+### Embedding span
+
+To trace an LLM span, specify the span kind as `embedding`, with optionally specifying the following arguments.
+
+**Note**: Annotating an embedding span's input requires different formatting than other span types. See [Annotating a span](#annotating-a-span) for more details on how to specify embedding inputs.
+
+#### Arguments
+
+`modelName`
+: required - _string_
+<br/>The name of the invoked LLM.
+
+`name`
+: optional - _string_
+<br/>The name of the operation. If not provided, `name` is set to the name of the traced function.
+
+`modelProvider`
+: optional - _string_ - **default**: `"custom"`
+
+`sessionId`
+: optional - _string_
+<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.
+
+`mlApp`
+: optional - _string_
+<br/>The name of the ML application that the operation belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information.
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function performEmbedding () {
+  ... // user application logic
+  return
+}
+performEmbedding = llmobs.wrap('embedding', { modelName: 'text-embedding-3', modelProvider: 'openai' }, performEmbedding)
+{{< /code-block >}}
+
+### Retrieval span
+
+To trace an LLM span, specify the span kind as `retrieval`, with optionally specifying the following arguments.
+
+**Note**: Annotating a retrieval span's output requires different formatting than other span types. See [Annotating a span](#annotating-a-span) for more details on how to specify retrieval outputs.
+
+#### Arguments
+
+`name`
+: optional - _string_
+<br/>The name of the operation. If not provided, `name` defaults to the name of the traced function.
+
+`sessionId`
+: optional - _string_
+<br/>The ID of the underlying user session. See [Tracking user sessions](#tracking-user-sessions) for more information.
+
+`mlApp`
+: optional - _string_
+<br/>The name of the ML application that the operation belongs to. See [Tracing multiple applications](#tracing-multiple-applications) for more information.
+
+#### Example
+
+{{< code-block lang="javascript" >}}
+function getRelevantDocs (question) {
+  const contextDocuments = ... // user application logic
+  llmobs.annotate({
+    inputData: question,
+    outputData: contextDocuments.map(doc => ({
+      id: doc.id,
+      score: doc.score,
+      text: doc.text,
+      name: doc.name
+    }))
+  })
+  return
+}
+getRelevantDocs = llmobs.wrap('retrieval', getRelevantDocs)
+{{< /code-block >}}
+
+### Conditions for finishing a span for a wrapped function
+
+`llmobs.wrap` extends the underlying behavior of `tracer.wrap`. The underlying span created when the function is called is finished under the following conditions:
```

Comment:
Can we link to `tracer.wrap()` documentation here for people that may not be APM customers?